neuralprophet/utils.py

```diff
@@ -987,3 +987,153 @@ def configure_trainer(
+
+        # Unpack additive event and holiday features
+        if "additive_events" in feature_indices:
+            events_start_idx, events_end_idx = feature_indices["additive_events"]
+            future_offset = max_lags - n_lags
+            inputs["events"] = OrderedDict()
+            inputs["events"]["additive"] = sliced_tensor[
+                :, future_offset : future_offset + n_forecasts + n_lags, events_start_idx : events_end_idx + 1
+            ]
+
+        # Unpack multiplicative event and holiday features
+        if "multiplicative_events" in feature_indices:
+            events_start_idx, events_end_idx = feature_indices["multiplicative_events"]
+            future_offset = max_lags - n_lags
+            if "events" not in inputs:
+                inputs["events"] = OrderedDict()
+            inputs["events"]["multiplicative"] = sliced_tensor[
+                :, future_offset : future_offset + n_forecasts + n_lags, events_start_idx : events_end_idx + 1
+            ]
+
+        # Unpack additive regressor features
+        if "additive_regressors" in feature_indices:
+            regressors_start_idx, regressors_end_idx = feature_indices["additive_regressors"]
+            future_offset = max_lags - n_lags
+            inputs["regressors"] = OrderedDict()
+            inputs["regressors"]["additive"] = sliced_tensor[
+                :,
+                future_offset : future_offset + n_forecasts + n_lags,
+                regressors_start_idx : regressors_end_idx + 1,
+            ]
+
+        # Unpack multiplicative regressor features
+        if "multiplicative_regressors" in feature_indices:
+            regressors_start_idx, regressors_end_idx = feature_indices["multiplicative_regressors"]
+            future_offset = max_lags - n_lags
+            if "regressors" not in inputs:
+                inputs["regressors"] = OrderedDict()
+            inputs["regressors"]["multiplicative"] = sliced_tensor[
+                :,
+                future_offset : future_offset + n_forecasts + n_lags,
+                regressors_start_idx : regressors_end_idx + 1,
+            ]
+
+        # Unpack seasonality feature
+        if config_seasonality is not None and hasattr(config_seasonality, "periods"):
+            inputs["seasonalities"] = OrderedDict()
+            for seasonality_name in config_seasonality.periods.keys():
+                seasonality_key = f"seasonality_{seasonality_name}"
+                if seasonality_key in feature_indices:
+                    seasonality_start_idx, seasonality_end_idx = feature_indices[seasonality_key]
+                    seasonality_offset = max_lags - n_lags
+                    inputs["seasonalities"][seasonality_name] = sliced_tensor[
+                        :,
+                        seasonality_offset : seasonality_offset + n_forecasts + n_lags,
+                        seasonality_start_idx:seasonality_end_idx,
```

Comment:
The previous features had +1 added to their end_index. Please check the comments in the stacking code and adjust here, too, after editing stacking to be consistent.